### ISSUE TITLE
[Vertical filter] Missing product list heading

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#1587]
## Issue summary

When a screen reader moves from the Filters into the product section, there is no notification of moving into the new section. This is typically handled via heading structure. Due to the DOM structure of vertical filters specifically, the heading notification is missing resulting in a confusing user experience.

Discovered while running usability testing with screen reader user.

#### HTML

```html
<facet-filters-form
  class="facets facets-vertical-sort page-width small-hide no-js-hidden"
>
  <form class="facets-vertical-form" id="FacetSortForm">
    <!-- … -->
    <h2 class="facet-filters__label caption-large text-body">
      <label for="SortBy">Sort by:</label>
    </h2>
    <!-- … -->
    <h2 class="product-count__text text-body">
      <span id="ProductCountDesktop">78 products</span>
    </h2>
    <!-- … -->
  </form>
</facet-filters-form>

<!-- … -->

<aside
  aria-labelledby="verticalTitle"
  class="facets-wrapper"
  id="main-collection-filters"
  data-id="template--15190262480918__product-grid"
>
  <!-- … -->
  <form id="FacetFiltersForm" class="facets__form-vertical">
    <!-- … -->
    <h2
      class="facets__heading facets__heading--vertical caption-large text-body"
      id="verticalTitle"
      tabindex="-1"
    >
      Filter:
    </h2>
    <!-- … -->
  </form>
  <!-- … -->
</aside>
```

#### Current structure
```
1. Sort [heading]
 └ Sort [content]

2. [n] Products [heading]

3. Filter [heading]
 └ Filter [content]
 └ Products [content]
```

## Steps to reproduce

1. Go to any Dawn demo store with any supported browser
2. Go to a product listing page with Vertical Filters enabled
3. With an active screen reader, use the virtual cursor to move from the Filters section to the product list section

### Expected

- Product heading to come **after** the filter/sort sections and **before** the product list.

### Actual

- Product heading comes **before** the filter section.

## Recommendation

1. Adjust the DOM to position the product heading directly before the product list.
2. Consider moving the Sort section directly after the `aside`, as seen in Horizontal filter setting layout.

This will provide a notification to screen reader users when moving from the filter section to the product list section.

#### HTML

```html
<aside
  aria-labelledby="verticalTitle"
  class="facets-wrapper"
  id="main-collection-filters"
  data-id="template--15190262480918__product-grid"
>
  <!-- … -->
  <form id="FacetFiltersForm" class="facets__form-vertical">
    <!-- … -->
    <h2
      class="facets__heading facets__heading--vertical caption-large text-body"
      id="verticalTitle"
      tabindex="-1"
    >
      Filter:
    </h2>
    <!-- … -->
  </form>
  <!-- … -->
  <facet-filters-form
  class="facets facets-vertical-sort page-width small-hide no-js-hidden"
>
  <form class="facets-vertical-form" id="FacetSortForm">
    <!-- … -->
    <h2 class="facet-filters__label caption-large text-body">
      <label for="SortBy">Sort by:</label>
    </h2>
    <!-- … -->
    <h2 class="product-count__text text-body">
      <span id="ProductCountDesktop">78 products</span>
    </h2>
    <!-- … -->
  </form>
</facet-filters-form>
</aside>
```

#### Recommended structure
```
1. Filter [heading]
 └ Filter [content]

2. Sort [heading]
 └ Sort [content]

3. [n] Products [heading]
 └ Products [content]
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Dawn version:** 5.0
- **Browser:** Any
- **Component:** Vertical filters
- **WCAG Principal:** Operable
- **WCAG SC:** [2.4.6 Headings and Labels](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html)
- **Severity:** Medium
- **Effort:** Medium
